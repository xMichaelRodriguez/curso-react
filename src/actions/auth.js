import Swal from "sweetalert2";
import { fetchAsync, fetchSync } from "../helpers/fetch";
import { types } from "../types/types";
import { eventLogout } from "./events";

export const startLogin = (email, password) => {
  return async (dispatch) => {
    Swal.fire({
      title: "Iniciando Sesion...",
      text: "Por favor espere...",
      allowOutsideClick: false,
      allowEnterKey: false,
      allowEscapeKey: false,
      showConfirmButton: false,
      onBeforeOpen: () => {
        Swal.showLoading();
      },
    });
    const resp = await fetchSync("auth", { email, password }, "POST");

    const body = await resp.json();

    if (body.ok) {
      localStorage.setItem("token", body.token);
      localStorage.setItem("token-initDate", new Date().getTime());
      dispatch(login({ uid: body.uid, name: body.name }));
      Swal.close();
    } else {
      Swal.close();
      Swal.fire("Error", body.msg, "error");
    }
  };
};

export const startRegister = (name, email, password) => {
  return async (dispatch) => {
    Swal.fire({
      title: "Registrando...",
      text: "Por favor espere...",
      allowOutsideClick: false,
      allowEnterKey: false,
      allowEscapeKey: false,
      showConfirmButton: false,
      onBeforeOpen: () => {
        Swal.showLoading();
      },
    });
    const resp = await fetchSync("auth/new", { name, email, password }, "POST");

    const body = await resp.json();

    if (body.ok) {
      localStorage.setItem("token", body.token);
      localStorage.setItem("token-initDate", new Date().getTime());
      dispatch(login({ uid: body.uid, name: body.name }));
      Swal.close();
    } else {
      Swal.close();
      Swal.fire("Error", body.msg, "error");
    }
  };
};

export const startChecking = () => {
  return async (dispatch) => {
    const resp = await fetchAsync("auth/renew");

    const body = await resp.json();

    if (body.ok) {
      localStorage.setItem("token", body.token);

      localStorage.setItem("token-initDate", new Date().getTime());

      dispatch(login({ uid: body.uid, name: body.name }));

      Swal.close();
    } else {
      dispatch(checkingFish());
    }
  };
};

const checkingFish = () => ({
  type: types.authFinishChecking,
});

const login = (user) => ({
  type: types.authLogin,
  payload: user,
});

export const startLogout = () => {
  return (dispatch) => {
    localStorage.clear();

    dispatch(eventLogout());

    dispatch(logout());
  };
};

const logout = () => ({
  type: types.authLogout,
});
