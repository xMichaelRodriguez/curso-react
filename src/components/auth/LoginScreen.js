import React from "react";
import { FaSignInAlt } from "react-icons/fa";
import { Link } from "react-router-dom";
export const LoginScreen = () => {
  const handleLogin = (e) => e.preventDefault();
  return (
    <div className="shadow auth">
      <form onSubmit={handleLogin}>
        <h1 className="mb-4">
          <ins>Blog</ins>
        </h1>

        <div className="form-row">
          <div className="col-md-12 mb-3">
            <div className="form-group">
              <label>Email</label>
              <input
                className="form-control"
                type="email"
                placeholder="example@dominio.com"
              />
            </div>
          </div>
          <div className="col-md-12 mb-3">
            <label>Password</label>
            <input
              type="password"
              className="form-control"
              placeholder="Password"
            />
          </div>
          <div className="col-md-6 mb-3">
            <button className="btn primary btn-lg" type="submit">
              <FaSignInAlt /> &nbsp;
              <span>Login</span>
            </button>
          </div>
          <div className="col-md-6 w-100">
            <Link to="/auth/register" class="btn btn-link text-capitalize">
              forgot password?
            </Link>
          </div>
        </div>
      </form>
    </div>
  );
};
