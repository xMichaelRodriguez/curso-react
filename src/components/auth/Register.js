import React from 'react';
import { FaSignInAlt } from 'react-icons/fa';
import { Link } from 'react-router-dom';

export const Register = () => {
  const handleRegister = (e) => e.preventDefault();
  return (
    <div className='shadow auth animate__animated animate__fadeIn'>
      <form onSubmit={handleRegister}>
        <h1 className='mb-4'>
          <ins>Blog</ins>
        </h1>

        <div className='form-row'>
          <div className='col-md-6'>
            <div className='form-group'>
              <label>First Name</label>
              <input
                className='form-control'
                type='text'
                placeholder='First Name'
              />
            </div>
          </div>
          <div className='col-md-6'>
            <div className='form-group'>
              <label>Last Name</label>
              <input
                className='form-control'
                type='text'
                placeholder='Last Name'
              />
            </div>
          </div>
          <div className='col-md-12 mb-3'>
            <div className='form-group'>
              <label>Email</label>
              <input
                className='form-control'
                type='email'
                placeholder='example@dominio.com'
              />
            </div>
          </div>
          <div className='col-md-12 mb-3'>
            <label>Password</label>
            <input
              type='password'
              className='form-control'
              placeholder='Password'
            />
          </div>
          <div className='col-md-6 mb-3'>
            <button className='btn primary btn-lg' type='submit'>
              <FaSignInAlt /> &nbsp;
              <span>Register</span>
            </button>
          </div>
          <div className='col-md-6 w-100'>
            <Link to='/auth/' class='btn btn-link text-capitalize'>
              I have an account
            </Link>
          </div>
        </div>
      </form>
    </div>
  );
};
