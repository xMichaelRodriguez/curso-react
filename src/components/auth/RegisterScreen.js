import React from "react";
import { Link } from "react-router-dom";

export const RegisterScreen = () => {
  return (
    <>
      <h3 className="auth__title mb-5">Register</h3>
      <form>
        <div className="auth__group">
          <input
            type="text"
            className="auth__input"
            name="firstName"
            placeholder="Your  First Name"
            autoComplete="false"
          />
          <span class="highlight"></span>
          <span class="bar"></span>
          <label>First Name</label>
        </div>
        <div className="auth__group">
          <input
            type="text"
            className="auth__input"
            name="lastName"
            placeholder="Your Last Name"
            autoComplete="false"
          />
          <span class="highlight"></span>
          <span class="bar"></span>
          <label>Last Name</label>
        </div>
        <div className="auth__group">
          <input
            type="email"
            className="auth__input"
            name="email"
            placeholder="Your Email"
            autoComplete="false"
          />
          <span class="highlight"></span>
          <span class="bar"></span>
          <label>Email</label>
        </div>
        <div className="auth__group">
          <input
            type="Password"
            className="auth__input"
            name="password"
            placeholder="Your Password"
          />
          <span class="highlight"></span>
          <span class="bar"></span>
          <label>Password</label>
        </div>

        <button type="submit" className="btn btn-primary btn-block mb-5">
          Sign In
        </button>

        <Link to="/auth/login" className="link">
          Already registered?
        </Link>
      </form>
    </>
  );
};
