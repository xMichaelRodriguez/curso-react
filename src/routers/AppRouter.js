import React from "react";

import {
  BrowserRouter as Router,
  Redirect,
  Route,
  Switch,
} from "react-router-dom";

import { JournalScren } from "../components/journal/JournalScren";
import { AuthRouter } from "./AuthRouter";
export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Switch>
          <Route path="/auth" component={AuthRouter} />
          <Route exact path="/" component={JournalScren} />

          <Redirect to="/auth/login" />
        </Switch>
      </div>
    </Router>
  );
};
