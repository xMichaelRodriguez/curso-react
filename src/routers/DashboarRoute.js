import React from "react";
import { Redirect, Route, Switch } from "react-router-dom";
import { DcScreen } from "../components/dc/DcScreen";
import { HeroesScreen } from "../components/heroes/HeroesScreen";
import { Navbar } from "../components/iu/NavBar";
import { MarvelScreen } from "../components/marvel/MarvelScreen";

export const DashboarRoute = () => {
  return (
    <>
      <Navbar />

      <div className="container mt-2">
        <Switch>
          <Route exact path="/marvel" component={MarvelScreen} />
          <Route exact path="/hero/:heroeId" component={HeroesScreen} />
          <Route exact path="/dc" component={DcScreen} />

          <Redirect to="/marvel" />
        </Switch>
      </div>
    </>
  );
};
