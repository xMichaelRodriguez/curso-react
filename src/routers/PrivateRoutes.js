import React from "react";
import PropTypes from "prop-types";

import { Redirect, Route } from "react-router-dom";

export const PrivateRoutes = ({
  isAuthenticated,
  render: Component,
  ...rest
}) => {
  localStorage.setItem('lastPath',rest.location.pathname)
  return (
    <Route
      {...rest}
      // render here!
      render={(props) =>
        isAuthenticated ? (
          <Component {...props} />
        ) : (
          <Redirect to="/login" />
        )
      }
    />
  );
};

PrivateRoutes.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  render: PropTypes.func.isRequired,
};
