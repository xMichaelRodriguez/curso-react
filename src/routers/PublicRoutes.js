import React from "react";
import PropTypes from "prop-types";

import { Redirect, Route } from "react-router-dom";

export const PublicRoutes = ({
  isAuthenticated,
  render: Component,
  ...rest
}) => {
  return (
    <Route
      {...rest}
      // render here!
      render={(props) =>
        !isAuthenticated ? (
          <Component {...props} />
        ) : (
          <Redirect to="/" />
        )
      }
    />
  );
};

PublicRoutes.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  render: PropTypes.func.isRequired,
};
