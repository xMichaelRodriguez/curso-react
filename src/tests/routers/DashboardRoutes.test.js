import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';
import { DashboardRoutes } from '../../routers/DashboardRoutes';

describe('Test in the <DashboardRoutes/>', () => {

    test('should display correctly', () => {
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: "MichaelRodriguez"
            }
        }
        const component = render(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        );
        expect(component).toMatchSnapshot();
        component.getByText('MichaelRodriguez')
    });

});
