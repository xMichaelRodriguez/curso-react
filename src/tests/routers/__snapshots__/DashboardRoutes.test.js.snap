// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in the <DashboardRoutes/> should display correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand-sm navbar-dark bg-dark"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Asociaciones
        </a>
        <div
          class="navbar-collapse"
        >
          <div
            class="navbar-nav"
          >
            <a
              aria-current="page"
              class="nav-item nav-link active"
              href="/marvel"
            >
              Marvel
            </a>
            <a
              class="nav-item nav-link"
              href="/dc"
            >
              DC
            </a>
            <a
              class="nav-item nav-link"
              href="/search"
            >
              Search
            </a>
          </div>
        </div>
        <div
          class="navbar-collapse collapse w-100 order-3 dual-collapse2"
        >
          <ul
            class="navbar-nav ml-auto"
          >
            <span
              class="nav-item nav-link text-white"
            >
              MichaelRodriguez
            </span>
            <button
              class="nav-item nav-link btn"
            >
              Logout
            </button>
          </ul>
        </div>
      </nav>
      <div
        class="container mt-2"
      >
        <div>
          <h1>
            Marvel Screen
          </h1>
          <hr />
          <div
            class="card-columns"
          >
            <div
              class="card ms-3 shadow animate__animated animate__fadeInUp"
              style="max-width: 540px;"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-md-4"
                >
                  <img
                    alt="Spider Man"
                    class="card-img"
                    src="./assets/heroes/marvel-spider.jpg"
                  />
                </div>
                <div
                  class="col-md-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                       
                      Spider Man
                       
                    </h5>
                    <p
                      class="card-text"
                    >
                       
                      Peter Parker
                       
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                         
                        Amazing Fantasy #15
                         
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-spider"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="card ms-3 shadow animate__animated animate__fadeInUp"
              style="max-width: 540px;"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-md-4"
                >
                  <img
                    alt="Captain America"
                    class="card-img"
                    src="./assets/heroes/marvel-captain.jpg"
                  />
                </div>
                <div
                  class="col-md-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                       
                      Captain America
                       
                    </h5>
                    <p
                      class="card-text"
                    >
                       
                      Steve Rogers
                       
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                         
                        Captain America Comics #1
                         
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-captain"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="card ms-3 shadow animate__animated animate__fadeInUp"
              style="max-width: 540px;"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-md-4"
                >
                  <img
                    alt="Iron Man"
                    class="card-img"
                    src="./assets/heroes/marvel-iron.jpg"
                  />
                </div>
                <div
                  class="col-md-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                       
                      Iron Man
                       
                    </h5>
                    <p
                      class="card-text"
                    >
                       
                      Tony Stark
                       
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                         
                        Tales of Suspense #39
                         
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-iron"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="card ms-3 shadow animate__animated animate__fadeInUp"
              style="max-width: 540px;"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-md-4"
                >
                  <img
                    alt="Thor"
                    class="card-img"
                    src="./assets/heroes/marvel-thor.jpg"
                  />
                </div>
                <div
                  class="col-md-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                       
                      Thor
                       
                    </h5>
                    <p
                      class="card-text"
                    >
                       
                      Thor Odinson
                       
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                         
                        Journey into Myster #83
                         
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-thor"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="card ms-3 shadow animate__animated animate__fadeInUp"
              style="max-width: 540px;"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-md-4"
                >
                  <img
                    alt="Hulk"
                    class="card-img"
                    src="./assets/heroes/marvel-hulk.jpg"
                  />
                </div>
                <div
                  class="col-md-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                       
                      Hulk
                       
                    </h5>
                    <p
                      class="card-text"
                    >
                       
                      Bruce Banner
                       
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                         
                        The Incredible Hulk #1
                         
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-hulk"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="card ms-3 shadow animate__animated animate__fadeInUp"
              style="max-width: 540px;"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-md-4"
                >
                  <img
                    alt="Wolverine"
                    class="card-img"
                    src="./assets/heroes/marvel-wolverine.jpg"
                  />
                </div>
                <div
                  class="col-md-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                       
                      Wolverine
                       
                    </h5>
                    <p
                      class="card-text"
                    >
                       
                      James Howlett
                       
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                         
                        The Incredible Hulk #180
                         
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-wolverine"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="card ms-3 shadow animate__animated animate__fadeInUp"
              style="max-width: 540px;"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-md-4"
                >
                  <img
                    alt="Daredevil"
                    class="card-img"
                    src="./assets/heroes/marvel-daredevil.jpg"
                  />
                </div>
                <div
                  class="col-md-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                       
                      Daredevil
                       
                    </h5>
                    <p
                      class="card-text"
                    >
                       
                      Matthew Michael Murdock
                       
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                         
                        Daredevil #1
                         
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-daredevil"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="card ms-3 shadow animate__animated animate__fadeInUp"
              style="max-width: 540px;"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-md-4"
                >
                  <img
                    alt="Hawkeye"
                    class="card-img"
                    src="./assets/heroes/marvel-hawkeye.jpg"
                  />
                </div>
                <div
                  class="col-md-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                       
                      Hawkeye
                       
                    </h5>
                    <p
                      class="card-text"
                    >
                       
                      Clinton Francis Barton
                       
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                         
                        Tales of Suspense #57
                         
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-hawkeye"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="card ms-3 shadow animate__animated animate__fadeInUp"
              style="max-width: 540px;"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-md-4"
                >
                  <img
                    alt="Cyclops"
                    class="card-img"
                    src="./assets/heroes/marvel-cyclops.jpg"
                  />
                </div>
                <div
                  class="col-md-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                       
                      Cyclops
                       
                    </h5>
                    <p
                      class="card-text"
                    >
                       
                      Scott Summers
                       
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                         
                        X-Men #1
                         
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-cyclops"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="card ms-3 shadow animate__animated animate__fadeInUp"
              style="max-width: 540px;"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-md-4"
                >
                  <img
                    alt="Silver Surfer"
                    class="card-img"
                    src="./assets/heroes/marvel-silver.jpg"
                  />
                </div>
                <div
                  class="col-md-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                       
                      Silver Surfer
                       
                    </h5>
                    <p
                      class="card-text"
                    >
                       
                      Norrin Radd
                       
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                         
                        The Fantastic Four #48
                         
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-silver"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Asociaciones
      </a>
      <div
        class="navbar-collapse"
      >
        <div
          class="navbar-nav"
        >
          <a
            aria-current="page"
            class="nav-item nav-link active"
            href="/marvel"
          >
            Marvel
          </a>
          <a
            class="nav-item nav-link"
            href="/dc"
          >
            DC
          </a>
          <a
            class="nav-item nav-link"
            href="/search"
          >
            Search
          </a>
        </div>
      </div>
      <div
        class="navbar-collapse collapse w-100 order-3 dual-collapse2"
      >
        <ul
          class="navbar-nav ml-auto"
        >
          <span
            class="nav-item nav-link text-white"
          >
            MichaelRodriguez
          </span>
          <button
            class="nav-item nav-link btn"
          >
            Logout
          </button>
        </ul>
      </div>
    </nav>
    <div
      class="container mt-2"
    >
      <div>
        <h1>
          Marvel Screen
        </h1>
        <hr />
        <div
          class="card-columns"
        >
          <div
            class="card ms-3 shadow animate__animated animate__fadeInUp"
            style="max-width: 540px;"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Spider Man"
                  class="card-img"
                  src="./assets/heroes/marvel-spider.jpg"
                />
              </div>
              <div
                class="col-md-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                     
                    Spider Man
                     
                  </h5>
                  <p
                    class="card-text"
                  >
                     
                    Peter Parker
                     
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                       
                      Amazing Fantasy #15
                       
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-spider"
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card ms-3 shadow animate__animated animate__fadeInUp"
            style="max-width: 540px;"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Captain America"
                  class="card-img"
                  src="./assets/heroes/marvel-captain.jpg"
                />
              </div>
              <div
                class="col-md-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                     
                    Captain America
                     
                  </h5>
                  <p
                    class="card-text"
                  >
                     
                    Steve Rogers
                     
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                       
                      Captain America Comics #1
                       
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-captain"
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card ms-3 shadow animate__animated animate__fadeInUp"
            style="max-width: 540px;"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Iron Man"
                  class="card-img"
                  src="./assets/heroes/marvel-iron.jpg"
                />
              </div>
              <div
                class="col-md-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                     
                    Iron Man
                     
                  </h5>
                  <p
                    class="card-text"
                  >
                     
                    Tony Stark
                     
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                       
                      Tales of Suspense #39
                       
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-iron"
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card ms-3 shadow animate__animated animate__fadeInUp"
            style="max-width: 540px;"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Thor"
                  class="card-img"
                  src="./assets/heroes/marvel-thor.jpg"
                />
              </div>
              <div
                class="col-md-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                     
                    Thor
                     
                  </h5>
                  <p
                    class="card-text"
                  >
                     
                    Thor Odinson
                     
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                       
                      Journey into Myster #83
                       
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-thor"
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card ms-3 shadow animate__animated animate__fadeInUp"
            style="max-width: 540px;"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Hulk"
                  class="card-img"
                  src="./assets/heroes/marvel-hulk.jpg"
                />
              </div>
              <div
                class="col-md-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                     
                    Hulk
                     
                  </h5>
                  <p
                    class="card-text"
                  >
                     
                    Bruce Banner
                     
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                       
                      The Incredible Hulk #1
                       
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-hulk"
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card ms-3 shadow animate__animated animate__fadeInUp"
            style="max-width: 540px;"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Wolverine"
                  class="card-img"
                  src="./assets/heroes/marvel-wolverine.jpg"
                />
              </div>
              <div
                class="col-md-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                     
                    Wolverine
                     
                  </h5>
                  <p
                    class="card-text"
                  >
                     
                    James Howlett
                     
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                       
                      The Incredible Hulk #180
                       
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-wolverine"
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card ms-3 shadow animate__animated animate__fadeInUp"
            style="max-width: 540px;"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Daredevil"
                  class="card-img"
                  src="./assets/heroes/marvel-daredevil.jpg"
                />
              </div>
              <div
                class="col-md-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                     
                    Daredevil
                     
                  </h5>
                  <p
                    class="card-text"
                  >
                     
                    Matthew Michael Murdock
                     
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                       
                      Daredevil #1
                       
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-daredevil"
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card ms-3 shadow animate__animated animate__fadeInUp"
            style="max-width: 540px;"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Hawkeye"
                  class="card-img"
                  src="./assets/heroes/marvel-hawkeye.jpg"
                />
              </div>
              <div
                class="col-md-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                     
                    Hawkeye
                     
                  </h5>
                  <p
                    class="card-text"
                  >
                     
                    Clinton Francis Barton
                     
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                       
                      Tales of Suspense #57
                       
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-hawkeye"
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card ms-3 shadow animate__animated animate__fadeInUp"
            style="max-width: 540px;"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Cyclops"
                  class="card-img"
                  src="./assets/heroes/marvel-cyclops.jpg"
                />
              </div>
              <div
                class="col-md-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                     
                    Cyclops
                     
                  </h5>
                  <p
                    class="card-text"
                  >
                     
                    Scott Summers
                     
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                       
                      X-Men #1
                       
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-cyclops"
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card ms-3 shadow animate__animated animate__fadeInUp"
            style="max-width: 540px;"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Silver Surfer"
                  class="card-img"
                  src="./assets/heroes/marvel-silver.jpg"
                />
              </div>
              <div
                class="col-md-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                     
                    Silver Surfer
                     
                  </h5>
                  <p
                    class="card-text"
                  >
                     
                    Norrin Radd
                     
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                       
                      The Fantastic Four #48
                       
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-silver"
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
